api/api/signals.py

```diff
@@ -1,8 +1,206 @@
+        logger.info("Getting installation UID...")
+        with open(INSTALLATION_UID_PATH, 'r') as file:
+            installation_uid = file.readline()
+    else:
+        logger.info("Populating installation UID...")
+        installation_uid = str(uuid.uuid4())
+        with open(INSTALLATION_UID_PATH, 'w') as file:
+            file.write(installation_uid)
+    app[INSTALLATION_UID_KEY] = installation_uid
+
+
+@cancel_signal_handler
+async def get_update_information(app: web.Application) -> None:
+    """Get updates information from Update Check Service and inject into the application context.
+
+    Parameters
+    ----------
+    app : web.Application
+        Application context to inject the update information.
+    """
+    current_version = f'v{_get_current_version()}'
+    headers = {
+        WAZUH_UID_KEY: app[INSTALLATION_UID_KEY],
+        WAZUH_TAG_KEY: current_version
+    }
+
+    async with aiohttp.ClientSession(connector=_get_connector()) as session:
+        while True:
+            logger.info('Getting updates information...')
+            logger.debug('Querying %s', RELEASE_UPDATES_URL)
+            try:
+                async with session.get(RELEASE_UPDATES_URL, headers=headers) as response:
+                    response_data = await response.json()
+
+                    logger.debug("Response status: %s", response.status)
+                    logger.debug("Response data: %s", response_data)
+
+                    update_information = {
+                        'last_check_date': get_utc_now(),
+                        'current_version': current_version,
+                        'status_code': response.status,
+                        'message': '',
+                        'last_available_major': {},
+                        'last_available_minor': {},
+                        'last_available_patch': {},
+                    }
+
+                    if response.status == 200:
+                        if len(response_data['data']['major']):
+                            update_information['last_available_major'].update(**response_data['data']['major'][-1])
+                        if len(response_data['data']['minor']):
+                            update_information['last_available_minor'].update(**response_data['data']['minor'][-1])
+                        if len(response_data['data']['patch']):
+                            update_information['last_available_patch'].update(**response_data['data']['patch'][-1])
+                    else:
+                        update_information['message'] = response_data['errors']['detail']
+
+                    app['update_information'] = update_information
+            except aiohttp.ClientError as err:
+                logger.error("Something went wrong when querying the update check service.", exc_info=err)
+            except Exception as err:
+                logger.error("An unknown error occurred while trying to get updates information.", exc_info=err)
```

Comment:
Showing the whole error traceback makes it look unhandled:
```
2023/10/09 09:57:34 INFO: Checking RBAC database integrity...
2023/10/09 09:57:34 INFO: /var/ossec/api/configuration/security/rbac.db file was detected
2023/10/09 09:57:34 INFO: RBAC database integrity check finished successfully
2023/10/09 09:57:39 INFO: Listening on 0.0.0.0:55000..
2023/10/09 09:57:39 INFO: Getting installation UID...
2023/10/09 09:57:39 INFO: Getting updates information...
2023/10/09 09:57:39 ERROR: Something went wrong when querying the update check service: Cannot connect to host cti:4041 ssl:default [Temporary failure in name resolution]
Traceback (most recent call last):
  File "/var/ossec/framework/python/lib/python3.9/site-packages/aiohttp/connector.py", line 1154, in _create_direct_connection
    hosts = await asyncio.shield(host_resolved)
  File "/var/ossec/framework/python/lib/python3.9/site-packages/aiohttp/connector.py", line 880, in _resolve_host
    addrs = await self._resolver.resolve(host, port, family=self._family)
  File "/var/ossec/framework/python/lib/python3.9/site-packages/aiohttp/resolver.py", line 33, in resolve
    infos = await self._loop.getaddrinfo(
  File "uvloop/loop.pyx", line 1507, in getaddrinfo
socket.gaierror: [Errno -3] Temporary failure in name resolution

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/var/ossec/framework/python/lib/python3.9/site-packages/api/signals.py", line 153, in get_update_information
    async with session.get(RELEASE_UPDATES_URL, headers=headers) as response:
  File "/var/ossec/framework/python/lib/python3.9/site-packages/aiohttp/client.py", line 1138, in __aenter__
    self._resp = await self._coro
  File "/var/ossec/framework/python/lib/python3.9/site-packages/aiohttp/client.py", line 535, in _request
    conn = await self._connector.connect(
  File "/var/ossec/framework/python/lib/python3.9/site-packages/aiohttp/connector.py", line 542, in connect
    proto = await self._create_connection(req, traces, timeout)
  File "/var/ossec/framework/python/lib/python3.9/site-packages/aiohttp/connector.py", line 907, in _create_connection
    _, proto = await self._create_direct_connection(req, traces, timeout)
  File "/var/ossec/framework/python/lib/python3.9/site-packages/aiohttp/connector.py", line 1166, in _create_direct_connection
    raise ClientConnectorError(req.connection_key, exc) from exc
aiohttp.client_exceptions.ClientConnectorError: Cannot connect to host cti:4041 ssl:default [Temporary failure in name resolution]
```

In my opinion, this is enough:
```suggestion
            except aiohttp.ClientError as err:
                logger.error(f"Something went wrong when querying the update check service: {err}")
            except Exception as err:
                logger.error(f"An unknown error occurred while trying to get updates information: {err}")
```
```
2023/10/09 09:58:08 INFO: Checking RBAC database integrity...
2023/10/09 09:58:08 INFO: /var/ossec/api/configuration/security/rbac.db file was detected
2023/10/09 09:58:08 INFO: RBAC database integrity check finished successfully
2023/10/09 09:58:13 INFO: Listening on 0.0.0.0:55000..
2023/10/09 09:58:13 INFO: Getting installation UID...
2023/10/09 09:58:13 INFO: Getting updates information...
2023/10/09 09:58:13 ERROR: Something went wrong when querying the update check service: Cannot connect to host cti:4041 ssl:default [Temporary failure in name resolution]
```